### DISCUSSION TITLE
Processing model for modification phase of profile resolution

### DISCUSSION THREAD

**Discussion by galtm** [usnistgov/OSCAL#1100] [Q&A]
I read a version of the profile specification from #1017 (before that PR was completed, so it was an interim version). In the section about add and remove directives in the modification phase, I didn't see an explanation of how directives were meant to be processed relative to each other.

A situation that may be similar is XQuery Update, whose processing model is described here:

* https://xmlmind.com/tutorials/XQueryUpdate/index.html (search for "pending update" and also see section 2.2)
* https://www.w3.org/TR/xquery-update-30/#id-precedence-rationale

To be clear, I'm not advocating for profile resolution's modification phase to act like XQuery Update. I just want to know how the profile alterations are meant to operate, such as:

1. Are add and remove directives processed in the sequence in which they appear in the profile, or in some other manner?
2. Are there rules about how the directives are allowed to relate to each other?

Can these matters be clarified in a future revision of the specification? Thanks!

**Comment by stephenbanghart** [comment]
This is good feedback, I'll take a look at the current draft and see if there are some places we can enhance clarity on this.
